File: generated-code/codecraft/swift/Sources/TransGenTest/BuildProperties.swift
public struct BuildProperties {
    let options: [EntityType]
    let initHealth: Int32?

    static func readFrom<S: InputStream>(_ stream: S) -> BuildProperties {
        var options: [EntityType]
        let optionsSize = stream.readInt32()
        options = (0..<optionsSize).map{ _ in
            var optionsSize: EntityType
            optionsSize = EntityType.readFrom(stream)
            return optionsSize
        }
        var initHealth: Int32?
        if stream.readBool() {
            initHealth = stream.readInt32()
        } else {
            initHealth = nil
        }
        return BuildProperties(options: options, initHealth: initHealth)
    }

    func writeTo<S: OutputStream>(_ stream: S) {
        stream.writeInt32(Int32(options.count))
        for optionsElement in options {
            optionsElement.writeTo(stream)
        }
        if initHealth == nil {
            stream.writeBool(false)
        } else {
            stream.writeBool(true)
            let initHealthValue = initHealth!
            stream.writeInt32(initHealthValue)
        }
    }
}

File: generated-code/cpp/codecraft/FileReadWrite/model/BuildProperties.cpp
#include "BuildProperties.hpp"

namespace model {

BuildProperties::BuildProperties(std::vector<model::EntityType> options, std::optional<int> initHealth) : options(options), initHealth(initHealth) { }

// Read BuildProperties from input stream
BuildProperties BuildProperties::readFrom(InputStream& stream) {
    std::vector<model::EntityType> options = std::vector<model::EntityType>();
    size_t optionsSize = stream.readInt();
    options.reserve(optionsSize);
    for (size_t optionsIndex = 0; optionsIndex < optionsSize; optionsIndex++) {
        model::EntityType optionsElement = readEntityType(stream);
        options.emplace_back(optionsElement);
    }
    std::optional<int> initHealth = std::optional<int>();
    if (stream.readBool()) {
        initHealth = stream.readInt();
    }
    return BuildProperties(options, initHealth);
}

// Write BuildProperties to output stream
void BuildProperties::writeTo(OutputStream& stream) const {
    stream.write((int)(options.size()));
    for (const model::EntityType& optionsElement : options) {
        stream.write((int)(optionsElement));
    }
    if (initHealth) {
        stream.write(true);
        const int& initHealthValue = *initHealth;
        stream.write(initHealthValue);
    } else {
        stream.write(false);
    }
}

// Get string representation of BuildProperties
std::string BuildProperties::toString() const {
    std::stringstream ss;
    ss << "BuildProperties { ";
    ss << "options: ";
    ss << "[ ";
    for (size_t optionsIndex = 0; optionsIndex < options.size(); optionsIndex++) {
        const model::EntityType& optionsElement = options[optionsIndex];
        if (optionsIndex != 0) {
            ss << ", ";
        }
        ss << entityTypeToString(optionsElement);
    }
    ss << " ]";
    ss << ", ";
    ss << "initHealth: ";
    if (initHealth) {
        const int& initHealthValue = *initHealth;
        ss << initHealthValue;
    } else {
        ss << "none";
    }
    ss << " }";
    return ss.str();
}

}